reports/executive_summary/run.sh

```diff
@@ -1,42 +1,107 @@
 #!/bin/bash
 
+USAGE="Usage: bash $0 {monthly|weekly} [report_start_yyyymmdd]\nIf not specified, report start defaults to the period ending yesterday"
 OUTPUT=output
-TODAY=$(date +%Y%m%d)
 if [ ! -d "$OUTPUT" ]; then
-    mkdir -p "$OUTPUT/sandbox_preservation"
+    mkdir -p "$OUTPUT"
 fi
 
-# If we have an argument, process that day.
-TARGET=$1
+# First argument is "mode". It is required.
+MODE=$1
+if [ "$MODE" != "weekly" -a "$MODE" != "monthly" ]; then
+    echo "Error: specify 'weekly' or 'monthly' report mode."
+    echo -e $USAGE
+    exit 1
+fi
+
+# If we have a date argument, use that as the report start day.
+TARGET=$2
 if [ -z "$TARGET" ]; then
-  # Default to processing "yesterday"
-  TARGET=$(date -d 'yesterday' +%Y%m%d)
+    # Default to processing the most recent completed reporting period.
+    # For a week, that is the period ending on the most recent Saturday (and
+    # starting on the prior Sunday)
+    # For a month, it is the period starting on the first of the previous month.
+    if [ "$MODE" = "weekly" ]; then
+        # The Sunday of the previous complete week
+        TARGET=$(date -d 'last sunday - 1 week' +%Y%m%d)
+    else
+        # The first day of the previous complete month
+        TARGET=$(date -d '1 month ago' +%Y%m01)
+    fi
 fi
 
-sed -r "s/__TARGET__/$TARGET/" schema_template.exec.json > schema.exec.json
+echo "Running $MODE report for period starting on $TARGET"
+
+# Make sure we have 'jq' and other prereqs
+export DEBIAN_FRONTEND=noninteractive; sudo apt-get --yes --force-yes install jq libpq-dev python-dev
+sudo pip install psycopg2
+
+# Fetch db connection details
+META=net-mozaws-prod-us-west-2-pipeline-metadata
+# Get metadata:
+aws s3 cp s3://$META/sources.json ./
+META_PREFIX=$(jq -r '.["telemetry-executive-summary-db"]["metadata_prefix"]' < sources.json)
+# Get read-only credentials:
+aws s3 cp s3://$META/$META_PREFIX/read/credentials.json ./
+
+DB_HOST=$(jq -r '.host' < credentials.json)
+DB_PORT=$(jq -r '.port' < credentials.json)
+DB_NAME=$(jq -r '.db_name' < credentials.json)
+DB_USER=$(jq -r '.username' < credentials.json)
+DB_PASS=$(jq -r '.password' < credentials.json)
+
+# Code expects a URL of the form:
+#   postgresql://username:password@hostname:port/dbname
+DB_URL="postgresql://${DB_USER}:${DB_PASS}@${DB_HOST}:${DB_PORT}/${DB_NAME}"
 
-echo "Fetching previous state..."
-aws s3 sync s3://telemetry-private-analysis-2/executive-report-v4/data/sandbox_preservation/ "$OUTPUT/sandbox_preservation/"
+CURRENT="$OUTPUT/executive_report.${MODE}.${TARGET}.csv"
+time python run_executive_report.py \
+        --verbose \
+        --check-tables \
+        --db-url "$DB_URL" \
+        --report-start $TARGET \
+        --mode $MODE > "$CURRENT"
+
+OVERALL="v4-${MODE}.csv"
+DASHBOARD_S3="s3://net-mozaws-prod-metrics-data/firefox-executive-dashboard"
+echo "Fetching previous state from $OVERALL..."
+aws s3 cp "$DASHBOARD_S3/$OVERALL" ./
```

Comment:
We should check the return codes on all these network operations (I have had them fail before).  This could assume no previous state when there actually was one.